### DISCUSSION TITLE
How to right align inner elements in HBox??

### DISCUSSION THREAD

**Discussion by cloudchamb3r** [fyne-io/fyne#3189] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/98369677/183110791-ab8b00ba-77fb-40d3-aa5a-ca40dacf6ffc.png)

```go
  func createLocalToNetworkContainer() *fyne.Container {
	  fileTree := createFileTree("~")
  
	  selectedLbl := widget.NewLabel("Selected : 0")
	  cancelBtn := widget.NewButton("Cancel", nil)
	  uploadBtn := widget.NewButton("Upload", nil)
	  btnBox := container.NewHBox(selectedLbl, cancelBtn, uploadBtn)
  
	  c := container.NewBorder(nil, btnBox, nil, nil, fileTree)
	  return c
  }
```
i'm new to fyne gui framework and i'm trying to implement simple view for my project.
i want to move move both cancel button and upload button to right side in HBox container
but in HBox container i couldn't find any align options or something else for inner elements..
how can i resolve this problem?

**Comment by Jacalz** [comment, marked as answer]
Hi there. You should be able to add a [spacer](https://pkg.go.dev/fyne.io/fyne/v2/layout#NewSpacer) inside the HBox like this:
```go
btnBox := container.NewHBox(selectedLbl, layout.NewSpacer(), cancelBtn, uploadBtn)
```
Another option would be to use a border layout with `selectedLbl` in the left position and a HBox containing `cancelBtn, uploadBtn` in the right position. Either option should give the desired effect.

**Reply by cloudchamb3r** [reply]
it works! awesome!!
thank you so much
